devel/nodejs14/Portfile

```diff
@@ -86,10 +83,36 @@ post-patch {
 
 # use the system libuv instead of the bundled version, as it is fixed for older systems
 if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
+    # on macOS nodejs only builds against libc++
+    # this force is OK as node does not link against any other c++ libs
+    # per @kencu, libcxx is only for 10.5 and 10.6 as a prerequisite for an even-newer libcxx (macports-libcxx)
+    depends_lib-append      port:libcxx
     depends_lib-append    path:lib/libuv.dylib:libuv
     configure.args-append --shared-libuv
 }
 
+#apparently even macos14 needs the flags that i originally included for older macs.
+#so i am going to make the 'ignore warnings as errors' applicable for all targets
+configure.cppflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+configure.cflags-append     -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+configure.cxxflags-append   -Wno-incompatible-function-pointer-types -Wno-enum-constexpr-conversion
+
+if { ${os.platform} eq "darwin" && ${os.major} < 15 } {
+    # macports-libcxx is required for all platforms below 10.13, i think.
+    # set it to brosemite and lower (14). increase to 18 if it indeed is up to high sierra.
+    depends_lib-append          port:macports-libcxx
```

Comment:
Isn’t this done via `legacysupport`?

Reply:
no. ken looked into this, see scroll above. macports legacysupport includes headers that conflict and break the build, though in theory the answer to your question is "yes"

Reply:
This it should be done one when the chosen runtime is `libc++`.

```
if {${os.platform} eq "darwin" && ${configure.cxx_stdlib} eq "libc++"} {
. . .
}
```

Reply:
Well, MacPorts’ node does, and that is wrong. But adding a wrong condition breaks things further, while setting it right does not hurt anyone.